lib/MusicBrainz/Server/Edit/Medium/SetTrackLengths.pm

```diff
@@ -63,6 +63,26 @@ sub foreign_keys {
     }
 }
 
+around _build_related_entities => sub {
+    my ($orig, $self) = splice(@_, 0, 2);
+    my $related = $self->$orig(@_);
+
+    my $medium = $self->c->model('Medium')->get_by_id($self->data->{medium_id});
+    $self->c->model('Track')->load_for_mediums($medium);
+
+    my $old_lengths = $self->data->{length}{old};
+    my $new_lengths = $self->data->{length}{new};
```

Comment:
`cdtoc_tracks` isn't cached, so I'd assign it to a variable first, otherwise it's recalculated for every track which itself loops over every track (n^2)

```suggestion
    my $new_lengths = $self->data->{length}{new};
    my $cdtoc_tracks = $medium->cdtoc_tracks;
```